opal/mca/accelerator/cuda/accelerator_cuda_component.c

```diff
@@ -115,30 +119,33 @@ static int accelerator_cuda_component_register(void)
     return OPAL_SUCCESS;
 }
 
-static opal_accelerator_base_module_t* accelerator_cuda_init(void)
+int opal_accelerator_cuda_delayed_init()
 {
-    int retval, i, j;
-    CUresult result;
+    CUresult result = OPAL_SUCCESS;
     CUcontext cuContext;
 
-    OBJ_CONSTRUCT(&opal_accelerator_cuda_stream_lock, opal_mutex_t);
+    /* Double checked locking to avoid having to
+     * grab locks post lazy-initialization.  */
+    if (true == accelerator_cuda_init_complete) {
```

Comment:
```suggestion
    opal_atomic_rmb();
    if (true == accelerator_cuda_init_complete) {
```
Establishes a synchronizes-with relationship with the memory barrier from the mutex unlock (release semantics)